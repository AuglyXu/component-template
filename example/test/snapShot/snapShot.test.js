import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

import SnapShot from './snapShot';

let container = null;
beforeEach(() => {
    // 创建一个 DOM 元素作为渲染目标
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    // 退出时进行清理
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it('应渲染问候语', () => {
    act(() => {
        render(<SnapShot />, container);
    });

    expect(pretty(container.innerHTML)).toMatchSnapshot(`"<span>嘿，陌生人</span>"`);

    act(() => {
        render(<SnapShot name="Jenny" />, container);
    });

    expect(pretty(container.innerHTML)).toMatchSnapshot(`"<h1>你好，Jenny！</h1>"`);

    act(() => {
        render(<SnapShot name="Margaret" />, container);
    });

    expect(pretty(container.innerHTML)).toMatchSnapshot(`"<h1>你好，Margaret！</h1>"`);
});
