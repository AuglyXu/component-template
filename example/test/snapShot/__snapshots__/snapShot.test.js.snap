// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`应渲染问候语: "<h1>你好，Jenny！</h1>" 1`] = `"<h1>你好，Jenny！</h1>"`;

exports[`应渲染问候语: "<h1>你好，Margaret！</h1>" 1`] = `"<h1>你好，Margaret！</h1>"`;

exports[`应渲染问候语: "<span>嘿，陌生人</span>" 1`] = `"<span>嘿，陌生人</span>"`;
